src/java/org/apache/cassandra/repair/RepairJob.java

```diff
@@ -125,13 +128,17 @@ public void run()
             {
                 // Request snapshot to all replica
                 List<Future<InetAddressAndPort>> snapshotTasks = new ArrayList<>(allEndpoints.size());
+                state.phase.snapshotsSubmitted();
                 for (InetAddressAndPort endpoint : allEndpoints)
                 {
-                    SnapshotTask snapshotTask = new SnapshotTask(desc, endpoint);
+                    SnapshotTask snapshotTask = new SnapshotTask(state.desc, endpoint);
```

Comment:
It's accessing `state.desc` all over the places. we can just leave `desc` outside of `JobState`.

Reply:
we expose this in the vtable, so will need.  You mentioned in https://github.com/apache/cassandra/pull/1419#discussion_r835780725 to move init of desc outside of the constructor, so if I do that then RepairJob can own it as well; will look to do this.